pkg/cmd/deployment/update/update.go

```diff
@@ -0,0 +1,154 @@
+package update
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"net/http"
+
+	"github.com/MakeNowJust/heredoc"
+	"github.com/cli/cli/v2/api"
+	"github.com/cli/cli/v2/internal/ghrepo"
+	"github.com/cli/cli/v2/pkg/cmdutil"
+	"github.com/cli/cli/v2/pkg/iostreams"
+	"github.com/spf13/cobra"
+)
+
+type updateOptions struct {
+	BaseRepo   func() (ghrepo.Interface, error)
+	HttpClient func() (*http.Client, error)
+	IO         *iostreams.IOStreams
+
+	Deployment     string
+	Status         string
+	LogURL         string
+	Description    string
+	Environment    string
+	EnvironmentURL string
+	AutoInactive   bool
+}
+
+func NewCmdUpdate(f *cmdutil.Factory, runF func(*updateOptions) error) *cobra.Command {
+	opts := updateOptions{
+		HttpClient: f.HttpClient,
+		IO:         f.IOStreams,
+	}
+
+	cmd := &cobra.Command{
+		Use:   "update [flags] <deployment>",
+		Short: "Update a deployment",
+		Long: heredoc.Docf(`
+			Update a deployment.
```

Comment:
I think there's some overlap here with https://github.com/cli/cli/pull/8718#discussion_r1574794480 and the point on a sub-command structure, e.g. `gh deployment status delete 132143234` though my gut feeling at the time was it might be one layer of commands too deep?

If not I suspect it'd be a simpler design to align to the Deployment API which would sort out the questions on how to add support for more `status` and `environment` endpoints.